tests/test_flow/test_api.py

```diff
@@ -376,6 +381,36 @@ def test_get_formats(preset_flow):
         assert list(p_series_index_df['q']) == [5]
 
 
+def test_get_modes_exclude_object(preset_flow):
+    # TODO: should we raise a warning for ValueProvider entities when
+    # queried for a mode that implies the existence of a persisted file?
+    flow = preset_flow
+
+    for mode in [Path, 'path']:
+        y_path = flow.get('y', mode=mode)
+        assert y_path is None
```

Comment:
Hmm, is this what we want?  Wouldn't it make more sense to throw an exception here?

Reply:
Added the following at the top of `get` to raise a `ValueError` for this situation:
```
        result_group = self._deriver.derive(name)
        if mode is object or mode == 'object':
            values = [result.value for result in result_group]
        else:
            # all other modes expect the entity to be persisted
            result_file_paths = [result.file_path for result in result_group]

            if None in result_file_paths:
                raise ValueError(
                    "Entity %r is not persisted but persisted file is expected by mode %r"
                    % (name, mode))
```